pkg/apps/deployment/status_test.go

```diff
@@ -0,0 +1,223 @@
+	)
+	tests := []struct {
+		name             string
+		pods             []*corev1.Pod
+		deployment       *appsv1.Deployment
+		expectedMessages []string
+		expectedError    error
+	}{
+		{
+			name: "ready with one restart",
+			pods: append([]*corev1.Pod{}, makePod(func(pod *corev1.Pod) {
+				pod.Labels = defaultLabels
+				pod.Status.Phase = corev1.PodRunning
+				pod.Status.ContainerStatuses = append(pod.Status.ContainerStatuses, corev1.ContainerStatus{
+					Name:         "test",
+					RestartCount: 1,
+					Ready:        true,
+				})
+			})),
+			deployment:       makeDeployment("foo", defaultLabels),
+			expectedMessages: []string{},
+		},
+		{
+			name: "not ready with one restart",
+			pods: append([]*corev1.Pod{}, makePod(func(pod *corev1.Pod) {
+				pod.Labels = defaultLabels
+				pod.Status.Phase = corev1.PodRunning
+				pod.Status.ContainerStatuses = append(pod.Status.ContainerStatuses, corev1.ContainerStatus{
+					Name:         "test",
+					RestartCount: 1,
+				})
+			})),
+			deployment:       makeDeployment("foo", defaultLabels),
+			expectedMessages: []string{"1 container is not ready in pod test"},
+		},
+		{
+			name: "not ready with two restarts",
+			pods: append([]*corev1.Pod{}, makePod(func(pod *corev1.Pod) {
+				pod.Labels = defaultLabels
+				pod.Status.Phase = corev1.PodRunning
+				pod.Status.ContainerStatuses = append(pod.Status.ContainerStatuses, corev1.ContainerStatus{
+					Name:         "test",
+					RestartCount: 2,
+				})
+			})),
+			deployment:       makeDeployment("foo", defaultLabels),
+			expectedMessages: []string{"1 container is crashlooping in pod test"},
+		},
+		{
+			name: "not ready and stucked in waiting",
+			pods: append([]*corev1.Pod{}, makePod(func(pod *corev1.Pod) {
+				pod.Labels = defaultLabels
+				pod.Status.Phase = corev1.PodPending
+				pod.Status.ContainerStatuses = append(pod.Status.ContainerStatuses, corev1.ContainerStatus{
+					Name: "test",
+					State: corev1.ContainerState{
+						Waiting: &corev1.ContainerStateWaiting{
+							Reason:  "IsSlow",
+							Message: "Slow container",
+						},
+					},
+				})
+			})),
+			deployment:       makeDeployment("foo", defaultLabels),
+			expectedMessages: []string{"1 container is waiting in pending pod test"},
+		},
+		{
+			name: "not ready stucked in waiting and crashlooping",
+			pods: append([]*corev1.Pod{}, makePod(func(pod *corev1.Pod) {
+				pod.Labels = defaultLabels
+				pod.Status.Phase = corev1.PodPending
+				pod.Status.ContainerStatuses = append(pod.Status.ContainerStatuses, corev1.ContainerStatus{
+					Name:         "test",
+					RestartCount: 5,
+					State: corev1.ContainerState{
+						Waiting: &corev1.ContainerStateWaiting{
+							Reason:  "IsSlow",
+							Message: "Slow container",
+						},
+					},
+				})
+			})),
+			deployment:       makeDeployment("foo", defaultLabels),
+			expectedMessages: []string{"1 crashlooping container is waiting in pending pod test"},
```

Comment:
`pod "test"`

Reply:
i hate quotes :)

Reply:
quotes where quotes make sense.

Reply:
this is a name. Either `pod %q` or `%s pod`. The current variant causes parse errors.

Reply:
ok :-) second variant then. fixed.